tests/trainer/optimization/test_parity_automatic_optimization.py

```diff
@@ -238,8 +245,62 @@ def test_lightning_optimizer_and_no_lightning_optimizer_equality_check_optim_cal
         assert mock_sgd_zero_grad.call_count == (expected_num_batches // accumulate_grad_batches)
         assert mock_sgd_step.call_count == mock_adam_step.call_count
         assert mock_sgd_step.call_count == mock_adam_step.call_count
-        assert mock_sgd_zero_grad.call_count == mock_adam_zero_grad.call_count
-        assert mock_sgd_zero_grad.call_count == mock_adamw_zero_grad.call_count
+
+
+def train_with_restore(tmpdir, model_cls, restore_from=None):
+    # init model
+    if restore_from is not None:
+        seed_everything(42)
+    model = model_cls(torch.optim.Adam, accumulate_grad_batches=1, lr=10e-1)
+    ckpt_saver = ModelCheckpoint(dirpath=f"{tmpdir}/mckpt", save_last=True, save_weights_only=False)
+    # Initialize a trainer
+    trainer = pl.Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=1 + bool(restore_from),
```

Comment:
```suggestion
        max_epochs=(1 + bool(restore_from)),
```

Reply:
won't `black` remove these parentheses?

Reply:
I hope not, the whitespace here is disturbing

Reply:
is it not correct?

Reply:
checked and no, it is not removed (thought it did remove redundant parentheses)